### PR TITLE
Cleanup and ensure that canUserJoinCall is live.

### PR DESCRIPTION
**Pull request by bmarty**
Small cleanup, see the commit messages for more details.

### CODE REVIEW

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt [hunk 1]
```diff
@@ -157,9 +155,9 @@ class MessagesPresenter @AssistedInject constructor(
             mutableStateOf(false)
         }
 
-        LaunchedEffect(currentSessionIdHolder.current) {
+        LaunchedEffect(syncUpdateFlow) {
```
Comment (ganfra) [inline features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt:158]: If you want the check to be done at each sync, you should use `syncUpdateFlow.value` instead

**Review by ganfra — APPROVED** [review]
One remark otherwise LGTM

File: features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt (hunk 1 above)
Comment (bmarty) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt:158]: Ah yes, good catch. So there is also a pb here https://github.com/element-hq/element-x-android/blob/develop/features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt#L168 ?

Comment (bmarty) [reply features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/MessagesPresenter.kt]: https://github.com/element-hq/element-x-android/pull/2313/commits/05cc526be98f08f4560c6a1795c091996bbcd2e6

**Pull request merged**